### PR TITLE
feat(types, js-sdk): add updateCustomer to js-sdk

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds new `POST /store/cart/:id/customer` endpoint to js sdk

### CODE REVIEW

File: packages/core/js-sdk/src/store/index.ts [hunk 1]
```diff
@@ -941,6 +941,38 @@ export class Store {
...
+     * @param headers - Headers to pass in the request.
+     * @returns The updated cart.
+     *
+     * @example
+     * sdk.store.cart.updateCustomer("cart_123", {})
+     * .then(({ cart }) => {
+     *   console.log(cart)
+     * })
+     */
+    updateCustomer: async (
```
Comment (olivermrbl) [inline packages/core/js-sdk/src/store/index.ts]: nit/suggestion(non-blocking): Wdyt of `transferCustomer`

File: packages/core/js-sdk/src/store/index.ts [hunk 2]
```diff
@@ -941,6 +941,38 @@ export class Store {
...
+      id: string,
+      body: HttpTypes.StoreUpdateCartCustomer,
```
Comment (olivermrbl) [inline packages/core/js-sdk/src/store/index.ts]: thought: Can we remove the `body` param and always pass `{}` in the `fetch` call?

That way you could consume it like so:
```ts
sdk.store.cart.transferCustomer("cart_123")
```

**Review by olivermrbl — APPROVED** [review]
LGTM, two suggestions :)

File: packages/core/js-sdk/src/store/index.ts (hunk 1 above)
Comment (riqwan) [reply packages/core/js-sdk/src/store/index.ts]: wouldn't it be the other way around? `transferCart('cart_id')`?

Comment (olivermrbl) [reply packages/core/js-sdk/src/store/index.ts]: Yes, my bad. Alternatively `setCustomer`. It's super nitpicky so I'll let you decide, but I just feel like `updateCustomer` intuitively tells me that I can update the customer information (not only the ID) on the cart 😄

File: packages/core/js-sdk/src/store/index.ts (hunk 2 above)
Comment (riqwan) [reply packages/core/js-sdk/src/store/index.ts]: done!

**Pull request merged**